### ISSUE TITLE
Way to interrupt recursive calls?

### ISSUE THREAD

**Issue by juggleru** [robertkrimen/otto#167]
Does anybody know way to interrupt recursive call like this?
In this sample interruption doesn't work.
```
package main

import (
	"github.com/robertkrimen/otto"
	"fmt"
	"errors"
	"time"
)

func Test() {
	runJSUnsafe("Test();")
}

func main() {

	val, err := runJSUnsafe("Test();")
	fmt.Println(val, err)
}

var halt = errors.New("SCTriggerTimeout")

func runJSUnsafe(code string) (result otto.Value, err error) {
	defer func() {
		if caught := recover(); caught != nil {
			if caught == halt {
				err = errors.New("Trigger executing time too long")
				return
			}
			err = errors.New(fmt.Sprintf("%#v", caught))
		}
	}()

	vm := otto.New()
	vm.Set("Test", Test)
	vm.Interrupt = make(chan func(), 1)
	go func() {
		time.Sleep(500 * time.Millisecond)
		vm.Interrupt <- func() {
			panic(halt)
		}
	}()
	result, err = vm.Run(code)
	return
}
```

**Comment by deoxxa** [comment]
This is just a plain old stack exhaustion situation. You'd get much the same result if you did something like this:

```go
package main
func a() { a() }
func main() { a() }
```

Otto doesn't do anything special regarding limiting call stack depth, but maybe there should be an option for that. So right now, no - there's nothing that you can do to prevent this particular problem. If you end up with some runaway recursive code, it will probably blow the stack out.

**Comment by deoxxa** [comment]
I've implemented a simple stack depth thing in #168. Can you take a look and see if it does what you want, @juggleru?

**Comment by juggleru** [comment]
@deoxxa oh thanks, it looks like all I need
